osf/models/preprint_provider.py

```diff
@@ -111,6 +113,20 @@ def absolute_api_v2_url(self):
         path = '/preprint_providers/{}/'.format(self._id)
         return api_v2_url(path)
 
+    def save(self, *args, **kwargs):
+        dirty_fields = self.get_dirty_fields()
+        old_id = dirty_fields.get('_id', None)
+        if old_id:
+            for permission_type in GROUPS.keys():
+                group_name = GROUP_FORMAT.format(provider_id=old_id, group=permission_type)
+                try:
+                    permission_group = Group.objects.get(name=group_name)
```

Comment:
Everything in this for loop could be simplified as 
```
Group.objects.filter(
    name=GROUP_FORMAT.format(provider_id=old_id, group=permissions_type
).update(
    name=GROUP_FORMAT.format(provider_id=self._id, group=permissions_type
)
```
This is conditional on `Group` not having anything important happening in `save`, however. Given that [it's just using](https://github.com/django/django/blob/master/django/contrib/auth/models.py#L94-L128) the default `Model.save`, and we're not updating either the primary key or foreign relations, it should be fine.